apps/studio/components/interfaces/SQLEditor/OngoingQueriesPanel.tsx

```diff
@@ -0,0 +1,133 @@
+import dayjs from 'dayjs'
+import { StopCircle } from 'lucide-react'
+import { useState } from 'react'
+import toast from 'react-hot-toast'
+
+import { useReadReplicasQuery } from 'data/read-replicas/replicas-query'
+import { useQueryAbortMutation } from 'data/sql/abort-query-mutation'
+import { useOngoingQueriesQuery } from 'data/sql/ongoing-queries-query'
+import { useSelectedProject } from 'hooks'
+import { useDatabaseSelectorStateSnapshot } from 'state/database-selector'
+import {
+  Button,
+  CodeBlock,
+  Sheet,
+  SheetContent,
+  SheetDescription,
+  SheetHeader,
+  SheetSection,
+  SheetTitle,
+  TooltipContent_Shadcn_,
+  TooltipTrigger_Shadcn_,
+  Tooltip_Shadcn_,
+  cn,
+} from 'ui'
+import ConfirmationModal from 'ui-patterns/Dialogs/ConfirmationModal'
+
+interface OngoingQueriesPanel {
+  visible: boolean
+  onClose: () => void
+}
+
+export const OngoingQueriesPanel = ({ visible, onClose }: OngoingQueriesPanel) => {
+  const project = useSelectedProject()
+  const state = useDatabaseSelectorStateSnapshot()
+  const [selectedId, setSelectedId] = useState<number>()
+
+  const { data: databases } = useReadReplicasQuery({ projectRef: project?.ref })
+  const database = (databases ?? []).find((db) => db.identifier === state.selectedDatabaseId)
+
+  const { data } = useOngoingQueriesQuery({
+    projectRef: project?.ref,
+    connectionString: database?.connectionString,
+  })
+  const queries = data ?? []
+
+  const { mutate: abortQuery, isLoading } = useQueryAbortMutation({
+    onSuccess: () => {
+      toast.success(`Successfully aborted query (ID: ${selectedId}`)
+      setSelectedId(undefined)
+    },
+  })
+
+  return (
+    <>
+      <Sheet open={visible} onOpenChange={() => onClose()}>
+        <SheetContent size="lg">
+          <SheetHeader>
+            <SheetTitle>
+              Running queries on{' '}
+              {database?.identifier === project?.ref ? 'primary database' : 'read replica'}
+            </SheetTitle>
+            <SheetDescription>
+              There {queries.length === 1 ? 'is' : 'are'}{' '}
+              <span className="text-foreground-light">{queries.length}</span> quer
+              {queries.length === 1 ? 'y' : 'ies'} currently running{' '}
+              {database?.identifier !== project?.ref ? `on replica ${database?.identifier}` : ''}
+            </SheetDescription>
+          </SheetHeader>
+          <div className="max-h-full h-full divide-y overflow-y-auto">
+            {queries.length === 0 && (
+              <div className="flex items-center justify-center h-full text-foreground-light text-sm">
+                No ongoing queries running on database
```

Comment:
A refresh button would be great somewhere in this component I reckon!

Reply:
yup good shout! will add that in